app/models/validations/setup_validations.rb

```diff
@@ -5,6 +5,33 @@ def validate_irproduct_other(record)
     end
   end
 
+  def validate_startdate(record)
+    if record.startdate > Time.zone.today
+      record.errors.add :startdate, I18n.t("validations.setup.startdate.today_or_earlier")
+    end 
+    
+    if record.voiddate.present?
+      if (record.startdate.to_date - record.voiddate.to_date).to_i.abs > 730
+        record.errors.add :startdate, I18n.t("validations.setup.startdate.voiddate_difference")
+      end 
+    end 
+
+    if record.mrcdate.present?
+      if (record.startdate.to_date - record.mrcdate.to_date).to_i.abs > 730
+        record.errors.add :startdate, I18n.t("validations.setup.startdate.mrcdate_difference")
+      end 
+    end 
+
+    if record.scheme_id.present?
+      scheme_end_date = Scheme.find(record.scheme_id).end_date
```

Comment:
Given we have an association do we need to do a lookup? Should be able to just do `record.scheme.end_date`?